packages/opentelemetry-resources/test/Resource.test.ts

```diff
@@ -99,6 +101,159 @@ describe('Resource', () => {
+      assert.ok(Resource.empty().asyncAttributesHaveResolved());
+    });
+  });
+
+  describe('asynchronous attributes', () => {
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should return true for asyncAttributesHaveResolved() if no promise provided', () => {
+      assert.ok(new Resource({ foo: 'bar' }).asyncAttributesHaveResolved());
+      assert.ok(Resource.empty().asyncAttributesHaveResolved());
+      assert.ok(Resource.default().asyncAttributesHaveResolved());
+    });
+
+    it('should return true for asyncAttributesHaveResolved() once promise finishes', async () => {
+      const clock = sinon.useFakeTimers();
+      const resourceResolve = new Resource(
+        {},
+        new Promise(resolve => {
+          setTimeout(resolve, 100);
+        })
+      );
+      const resourceReject = new Resource(
+        {},
+        new Promise((_, reject) => {
+          setTimeout(reject, 200);
+        })
+      );
+
+      for (const resource of [resourceResolve, resourceReject]) {
+        assert.ok(!resource.asyncAttributesHaveResolved());
+        await clock.nextAsync();
+        await resource.waitForAsyncAttributes();
+        assert.ok(resource.asyncAttributesHaveResolved());
+      }
+    });
+
+    it('should merge async attributes into sync attributes once resolved', async () => {
+      //async attributes that resolve after 1 ms
+      const asyncAttributes = new Promise<ResourceAttributes>(resolve => {
+        setTimeout(
+          () => resolve({ async: 'fromasync', shared: 'fromasync' }),
+          1
+        );
+      });
+
+      const resource = new Resource(
+        { sync: 'fromsync', shared: 'fromsync' },
+        asyncAttributes
+      );
+
+      await resource.waitForAsyncAttributes();
+      assert.deepStrictEqual(resource.attributes, {
+        sync: 'fromsync',
+        // async takes precedence
+        shared: 'fromasync',
+        async: 'fromasync',
+      });
+    });
+
+    it('should merge async attributes when both resources have promises', async () => {
+      const resource1 = new Resource(
+        {},
+        Promise.resolve({ promise1: 'promise1val', shared: 'promise1val' })
+      );
+      const resource2 = new Resource(
+        {},
+        Promise.resolve({ promise2: 'promise2val', shared: 'promise2val' })
+      );
+      // this one rejects
+      const resource3 = new Resource({}, Promise.reject(new Error('reject')));
+      const resource4 = new Resource(
+        {},
+        Promise.resolve({ promise4: 'promise4val', shared: 'promise4val' })
+      );
+
+      const merged = resource1
+        .merge(resource2)
+        .merge(resource3)
+        .merge(resource4);
+      await merged.waitForAsyncAttributes();
+      assert.deepStrictEqual(merged.attributes, {
+        promise1: 'promise1val',
+        promise2: 'promise2val',
+        promise4: 'promise4val',
+        // same behavior as for synchronous attributes
```

Comment:
which behavior are you referring to?

Reply:
Refers to mering behavior. Im actually going to remove this since it states nothing other than the obvious